examples/knative/powercli/kn-pcli-snapshot-cron/handler.ps1

```diff
@@ -0,0 +1,180 @@
+   try {
+      Connect-VIServer -Server $VCENTER_SERVER -User $VCENTER_USERNAME -Password $VCENTER_PASSWORD
+   } catch {
+      Write-Error "$(Get-Date) - Failed to connect to vCenter Server"
+      throw $_
+   }
+
+   Write-Host "$(Get-Date) - Successfully connected to $VCENTER_SERVER`n"
+
+   Write-Host "$(Get-Date) - Init Processing Completed`n"
+}
+
+Function Process-Shutdown {
+   [CmdletBinding()]
+   param()
+   Write-Host "$(Get-Date) - Processing Shutdown`n"
+
+   Write-Host "$(Get-Date) - Disconnecting from vCenter Server`n"
+
+   try {
+      Disconnect-VIServer * -Confirm:$false
+   } catch {
+      Write-Error "$(Get-Date) - Failed to Disconnect from vCenter Server"
+   }
+
+   Write-Host "$(Get-Date) - Shutdown Processing Completed`n"
+}
+
+Function Process-Handler {
+   [CmdletBinding()]
+   param(
+      [Parameter(Position=0,Mandatory=$true)][CloudNative.CloudEvents.CloudEvent]$CloudEvent
+   )
+
+   # Decode CloudEvent
+   try {
+      $cloudEventData = $cloudEvent | Read-CloudEventJsonData -Depth 10
+   } catch {
+      throw "`nPayload must be JSON encoded"
+   }
+
+   if(${env:FUNCTION_DEBUG} -eq "true") {
+      Write-Host "$(Get-Date) - DEBUG: CloudEvent`n $(${cloudEvent} | Out-String)`n"
+
+      Write-Host "$(Get-Date) - DEBUG: CloudEventData`n $(${cloudEventData} | ConvertTo-Json)`n"
+   }
+
+   $ProgressPreference = "SilentlyContinue"
+
+   # Extract snapshot retention configuration & VM list from 'data' payload
+   $maxSnapshotCount = ${cloudEventData}.retentionConfig.count
+   $maxSnapshotSize = ${cloudEventData}.retentionConfig.sizeGB
+   $maxSnapshotDays = ${cloudEventData}.retentionConfig.days
+   $vmList = ${cloudEventData}.virtualMachines
+   $dryRun = ${cloudEventData}.dryRun
+
+   # Check to ensure at least one snapshot policy is defined
+   if($maxSnapshotCount -eq "" -and $maxSnapshotSize -eq "" -and $maxSnapshotDays -eq "") {
+      throw "No snapshot retentation policy were defined"
+   }
+
+   # Check to ensure we have a list of VMs
+   if($vmList -eq "" -or $vmList.getType().BaseType.Name -ne "Array") {
+      throw "Invalid or missing input found for VM list"
+   }
+
+   # Retrieve list of VMs to run snapshot management against
+   try {
+      $vms = Get-Vm -NoRecursion -Name $vmList
+   } catch {
+      throw "Failed to retrieve VM list"
+   }
+
+   foreach ($vm in $vms) {
+      $vmName = ${vm}.name
+
+      Write-Host "$(Get-Date) - Checking VM: $vmName "
+      try {
+         $snapshots = $vm | Get-Snapshot
+      } catch {
+         throw "Failed to retrieve snapshots from VM $vmName "
+      }
+
+      if($maxSnapshotSize -ne "") {
+         foreach ($snapshot in $snapshots) {
+            $snapshotName = ${snapshot}.name
+
+            # Get snapshot size (GB) and check against max size
+            if($snapshot.sizeGB -gt $maxSnapshotSize) {
+               Write-Host "$(Get-Date) - `tSnapshot $snapshotName consumes $(${snapshot}.SizeGB) GB and exceeds maximum size (${maxSnapshotSize} GB)"
```

Comment:
`t < for tab?

Reply:
Yup, '`' is PS special escape character :)